core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -638,10 +640,21 @@ public void onClosed(Channel c, IOException cause) {
         String slaveVersion = channel.call(new SlaveVersion());
         log.println("Remoting version: " + slaveVersion);
         VersionNumber agentVersion = new VersionNumber(slaveVersion);
+        boolean rejectConnection = GlobalConfiguration.all().get(RemotingConfiguration.class).getRejectConnection();
+
+
         if (agentVersion.isOlderThan(RemotingVersionInfo.getMinimumSupportedVersion())) {
-            log.println(String.format("WARNING: Remoting version is older than a minimum required one (%s). " +
-                    "Connection will not be rejected, but the compatibility is NOT guaranteed",
-                    RemotingVersionInfo.getMinimumSupportedVersion()));
+            if (rejectConnection) {
+                log.println(String.format("Remoting version is older than a minimum required one (%s). " +
+                                "Connection is rejected.",
+                        RemotingVersionInfo.getMinimumSupportedVersion()));
+                channel.close();
```

Comment:
This closes the channel, however the event for connecting agent to Jenkins keeps getting triggered. This leads to a sort of loop. It would be really helpful if somebody could advice me on how to break this loop.